client/state/themes/selectors.js

```diff
@@ -0,0 +1,200 @@
+}
+
+/**
+ * Returns the URL for a given theme's setup instructions
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Object}  theme  Theme object
+ * @param  {?Number} siteId Site ID to optionally use as context
+ * @return {?String}        Theme setup instructions URL
+ */
+export function getThemeSupportUrl( state, theme, siteId ) {
+	if ( ! theme || ! isPremiumTheme( theme ) ) {
+		return null;
+	}
+
+	const sitePart = siteId ? `/${ getSiteSlug( state, siteId ) }` : '';
+
+	if ( config.isEnabled( 'manage/themes/details' ) ) {
+		return `/theme/${ theme.id }/setup${ sitePart }`;
+	}
+
+	return `${ oldShowcaseUrl }${ sitePart }${ theme.id }/support`;
+}
+
+/**
+ * Returns the URL for a given theme's support page.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Object}  theme  Theme object
+ * @param  {?Number} siteId Site ID to optionally use as context
+ * @return {?String}        Theme support page URL
+ */
+export function getThemeHelpUrl( state, theme, siteId ) {
+	if ( ! theme ) {
+		return null;
+	}
+
+	if ( isJetpackSite( state, siteId ) ) {
+		return '//wordpress.org/support/theme/' + theme.id;
+	}
+
+	let baseUrl = oldShowcaseUrl + theme.id;
+	if ( config.isEnabled( 'manage/themes/details' ) ) {
+		baseUrl = `/theme/${ theme.id }/support`;
+	}
+
+	return baseUrl + ( siteId ? `/${ getSiteSlug( state, siteId ) }` : '' );
+}
+
+/**
+ * Returns the URL for purchasing the given theme for the given site.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Object}  theme  Theme object
+ * @param  {Number}  siteId Site ID for which to buy the theme
+ * @return {?String}        Theme purchase URL
+ */
+export function getThemePurchaseUrl( state, theme, siteId ) {
+	if ( ! isPremiumTheme( theme ) ) {
+		return null;
+	}
+
+	return `/checkout/${ getSiteSlug( state, siteId ) }/theme:${ theme.id }`;
+}
+
+/**
+ * Returns the URL for opening the customizer with the given theme on the given site.
+ *
+ * @param  {Object}   state  Global state tree
+ * @param  {?Object}  theme  Theme object
+ * @param  {?Number}  siteId Site ID to open the customizer for
+ * @return {?String}         Customizer URL
+ */
+export function getThemeCustomizeUrl( state, theme, siteId ) {
+	if ( ! siteId ) {
+		return '/customize/';
+	}
+
+	if ( isJetpackSite( state, siteId ) ) {
+		return getSiteOption( state, siteId, 'admin_url' ) +
+			'customize.php?return=' +
+			encodeURIComponent( window.location ) +
+			( theme ? '&theme=' + theme.id : '' );
+	}
+
+	return '/customize/' + getSiteSlug( state, siteId ) + ( theme && theme.stylesheet ? '?theme=' + theme.stylesheet : '' );
+}
+
+/**
+ * Returns the URL for signing up for a new WordPress.com account with the given theme pre-selected.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Object}  theme  Theme object
+ * @return {?String}        Signup URL
+ */
+export function getThemeSignupUrl( state, theme ) {
```

Comment:
We are not using `state` anywhere in the body of this selector. Can we drop it? Or is it better to keep it just for consistency sake?